src/processor/operator/persistent/reader/rdf/rdf_reader.cpp

```diff
@@ -262,22 +253,24 @@ offset_t RdfReader::read(DataChunk* dataChunk) {
     } break;
     }
     vectorSize = 0;
-    rowOffset = 0;
     while (true) {
         status = serd_reader_read_chunk(reader);
+        if (vectorSize > DEFAULT_VECTOR_CAPACITY) {
+            printf("warning\n");
+        }
         if (status == SERD_ERR_BAD_SYNTAX) {
             // Skip to the next line.
             serd_reader_skip_until_byte(reader, (uint8_t)'\n');
             continue;
         }
         // Each line from turtle will increase vector size by 3. To avoid exceeding
         // DEFAULT_VECTOR_CAPACITY, we check the size by adding 3 first.
-        if (status != SERD_SUCCESS || vectorSize + 3 >= DEFAULT_VECTOR_CAPACITY) {
+        if (status != SERD_SUCCESS || vectorSize + 100 >= DEFAULT_VECTOR_CAPACITY) {
```

Comment:
Where is 100 coming from? Can you also update the comment `increase vector size by 3`.